cupy/fft/fft.py

```diff
@@ -83,14 +83,27 @@ def _exec_fft(a, direction, value_type, norm, axis, overwrite_x,
         out_size = a.shape[-1]
 
     batch = a.size // a.shape[-1]
-    plan = cufft.Plan1d(out_size, fft_type, batch)
+    if plan is None:
+        plan = cufft.Plan1d(out_size, fft_type, batch)
+    else:
+        # check plan validity
+        if not isinstance(plan, cufft.Plan1d):
+            raise ValueError("expected plan to have type cufft.Plan1d")
+        if fft_type != plan.fft_type:
+            raise ValueError("CUFFT plan dtype mismatch.")
+        if out_size != plan.nx:
+            raise ValueError("Target array size mismatches the plan.")
+        if batch != plan.batch:
+            raise ValueError("Batch size mismatches the plan.")
```

Comment:
same as previous comment regarding "mismatches the"

Reply:
Done.